src/audio/module_adapter/module_adapter.c

```diff
@@ -596,38 +606,52 @@ int module_adapter_copy(struct comp_dev *dev)
 	struct processing_module *mod = comp_get_drvdata(dev);
 	struct module_data *md = &mod->priv;
 	struct comp_buffer *source, *sink;
-	struct comp_buffer __sparse_cache *source_c = NULL, *sink_c = NULL;
+	struct comp_buffer __sparse_cache *source_c[PLATFORM_MAX_STREAMS];
+	struct comp_buffer __sparse_cache *sink_c = NULL;
 	struct comp_copy_limits c;
 	struct list_item *blist;
 	size_t size = MAX(mod->deep_buff_bytes, mod->period_bytes);
 	uint32_t min_free_frames = UINT_MAX;
+	uint32_t num_input_buffers = 0;
 	int ret, i = 0;
 
 	comp_dbg(dev, "module_adapter_copy(): start");
 
 	/*
 	 * Simplify calculation of bytes_to_process for modules that produce period_bytes every
-	 * period and have only 1 source and 1 sink buffer
+	 * period and have N sources and only 1 sink buffer
 	 */
 	if (mod->simple_copy) {
-		source = list_first_item(&dev->bsource_list, struct comp_buffer, sink_list);
 		sink = list_first_item(&dev->bsink_list, struct comp_buffer, source_list);
-
-		source_c = buffer_acquire(source);
 		sink_c = buffer_acquire(sink);
-		comp_get_copy_limits_frame_aligned(source_c, sink_c, &c);
+		list_for_item(blist, &dev->bsource_list) {
+			source = container_of(blist, struct comp_buffer, sink_list);
+			source_c[i] = buffer_acquire(source);
 
-		buffer_stream_invalidate(source_c, c.frames * c.source_frame_bytes);
+			mod->input_buffers[i].size = 0;
 
-		/* note that the size is in number of frames not the number of bytes */
-		mod->input_buffers[0].size = c.frames;
-		mod->input_buffers[0].consumed = 0;
-		mod->output_buffers[0].size = 0;
+			/* check if the source dev is in the same state as the dev */
+			if (!source_c[i]->source || source_c[i]->source->state != dev->state)
+				continue;
+
+			num_input_buffers++;
+
+			comp_get_copy_limits_frame_aligned(source_c[i], sink_c, &c);
+
+			buffer_stream_invalidate(source_c[i], c.frames * c.source_frame_bytes);
 
-		mod->input_buffers[0].data = &source_c->stream;
+			/* note that the size is in number of frames not the number of bytes */
+			mod->input_buffers[i].size = c.frames;
+			mod->input_buffers[i].consumed = 0;
+
+			mod->input_buffers[i].data = &source_c[i]->stream;
+			i++;
```

Comment:
num_input_buffers and i are doing the same count

Reply:
this is a good catch. They shouldn't be. I've moved the i++ now.